### DISCUSSION TITLE
Handling events of a open62541 server

### DISCUSSION THREAD

**Discussion by clashman** [FreeOpcUa/python-opcua#1424] [Q&A]
I am using this server example:
https://github.com/open62541/open62541/blob/master/examples/tutorial_server_events.c

And this client example:
https://github.com/FreeOpcUa/python-opcua/blob/master/examples/client-events.py
With only line 43 and 46 changed to make it compatible:
obj = client.get_server_node()
myevent = root.get_child(["0:Types", "0:EventTypes", "0:BaseEventType", "0:SimpleEventType"])

When I trigger the event, the server sends this error:
**opcua.ua.uaerrors._auto.BadFilterOperandCountMismatch: "The number of operands provided for the filter operator was less then expected for the operand provided."(BadFilterOperandCountMismatch)**

What is going on there?

**Comment by oroulet** [comment]
I think someone reported a similar issue a while ago. Try to search. I do not guarantee or implementation is bug free. But the openxxx guys have also their bugs

**Comment by derigo** [comment]
I have the same issue, the only thing i can say is that my open62541 server events are working in UAExpert correctly

**Reply by clashman** [reply]
If I remember correctly, my issue was due to fact that open62541 did not yet have event filter support at that time(prior to v1.3). I circumvented this on the open62541-side by changing the source code to simply ignore filters instead of replying with an error. Nowadays, open62541 has event filter support, so it should work out of the box.

**Reply by derigo** [reply]
Thank you for pointing this out, i updated open62541 and debugged the subscription code in the open62541 library, the subscripton now seems to work in the open62541 part but i still cannot receive the custom event in my python handler in asyncua  ( .. but this works in ua expert) , i attach my custom client code, i had to do the subsription with an evfilter object to get it to work without error in the client. 

[client_test.txt](https://github.com/FreeOpcUa/python-opcua/files/10402764/client_test.txt)

**Reply by derigo** [reply]
Finally i got it working! 
It was my mistake, i did a while loop at the end with a sleep in it => i had to use an " await asyncio.sleep" instead to make it work

instead of 
` while True: 
                sleep(0.1)`

write 
` while True: 
                await asyncio.sleep(0.1)`

that fixed the issue!

i appeded the working example 








[client_test.txt](https://github.com/FreeOpcUa/python-opcua/files/10828658/client_test.txt)
